gen2-record-replay/libraries/depthai_record.py

```diff
@@ -31,43 +39,63 @@ def __init__(self, path: Path, device) -> None:
         calibData.eepromToJsonFile(str(self.path / "calib.json"))
 
         self.convert_mp4 = False
+        self.mcap = mcap
 
     def run(self):
         files = {}
+        mcaps = {}
+
         def create_video_file(name):
-            if name == 'depth': # or (name=='color' and 'depth' in self.save):
+            if name == 'depth':  # or (name=='color' and 'depth' in self.save):
                 files[name] = self.depthAiBag
             else:
                 ext = 'h265' if self.quality == EncodingQuality.LOW else 'mjpeg'
                 files[name] = open(str(self.path / f"{name}.{ext}"), 'wb')
             # if name == "color": fourcc = "I420"
             # elif name == "depth": fourcc = "Y16 " # 16-bit uncompressed greyscale image
             # else : fourcc = "GREY" #Simple, single Y plane for monochrome images.
-            # files[name] = VideoWriter(str(path / f"{name}.avi"), VideoWriter_fourcc(*fourcc), fps, sizes[name], isColor=name=="color")
+            # files[name] = VideoWriter(str(path / f"{name}.avi"), VideoWriter_fourcc(*fourcc), fps, sizes[name],
+            # isColor=name=="color")
+
+        def create_mcap_file(name):
+            mcaps[name] = Mcap(str(self.path / f"{name}"))
+            mcaps[name].imageInit()
 
         while True:
             try:
                 frames = self.frame_q.get()
                 if frames is None:
                     break
                 for name in frames:
-                    if name not in files: # File wasn't created yet
-                        create_video_file(name)
+                    if not self.mcap:
+                        if name not in files:  # File wasn't created yet
+                            create_video_file(name)
+                        # if self.rotate != -1:  # Doesn't work atm
+                        # frames[name] = cv2.rotate(frames[name], self.rotate)
+
+                        files[name].write(frames[name])
+
+                        # frames[name].tofile(files[name])
+
+                    else:
+                        if name not in mcaps:  # File wasn't created yet
+                            create_mcap_file(name)
 
-                    # if self.rotate != -1: # Doesn't work atm
-                    #     frames[name] = cv2.rotate(frames[name], self.rotate)
+                        mcaps[name].imageSave(frames[name])
```

Comment:
Can't all streams be saved in a single MCAP file? If that is possible, I assume schemas will get a bit more complicated (as you would need to dynamically create it), but we should go for that.